splice_si_api/src/main/java/com/splicemachine/si/impl/server/SITransactor.java

```diff
@@ -574,72 +588,142 @@ private ConflictResults checkCommitTimestampForConflict(TxnView updateTransactio
-                    }
-                    throw exceptionLib.writeWriteConflict(dataTransactionId,updateTransaction.getTxnId());
-            }
+                }
         }
+
         return conflictResults;
     }
 
     // Helpers
 
+    private long[] getDataActiveTransactions(TxnView txn, Partition table, KVPair kvPair, TxnSupplier txnSupplier, DataCell cell) throws IOException {
+        if(shouldIgnoreConflicts(table)) {
+            if(LOG.isTraceEnabled()) {
+                LOG.trace(String.format("detection of conflicting data active txns ignored for txn %d as it affects table %s",
+                                        txn.getTxnId(),
+                                        table.getTableName()));
+            }
+            return null;
+        }
+        DataGet dataGet = opFactory.newGet(kvPair.getRowKey(), null);
+        dataGet.addColumn(SIConstants.DEFAULT_FAMILY_BYTES, SIConstants.TOMBSTONE_COLUMN_BYTES);
+        dataGet.addColumn(SIConstants.DEFAULT_FAMILY_BYTES, SIConstants.PACKED_COLUMN_BYTES);
+        if(kvPair.getType() == KVPair.Type.DELETE) {
+            dataGet.addColumn(SIConstants.DEFAULT_FAMILY_BYTES, SIConstants.FK_COUNTER_COLUMN_BYTES);
+        }
+        dataGet.returnAllVersions();
+        DataResult result = table.get(dataGet, null);
+        Iterator<DataCell> iterator = result.iterator();
+        Set<Long> activeTx = new HashSet<>();
+        while(iterator.hasNext()) {
+            long dataTxnId = iterator.next().version();
+            if((dataTxnId & SIConstants.TRANSANCTION_ID_MASK) == txn.getTxnId()) {
+                continue;
+            }
+            // get the parent
+            TxnView txnView = txnSupplier.getTransaction(dataTxnId);
+            while(true) {
+                TxnView parent = txnView.getParentTxnView();
+                if(parent == Txn.ROOT_TRANSACTION) break;
+                txnView = parent;
+            }
+            Txn.State state = txnView.getState();
+            if(state == Txn.State.ACTIVE) {
+                activeTx.add(txnView.getTxnId() & SIConstants.TRANSANCTION_ID_MASK);
+            } else if(state == Txn.State.COMMITTED && txnView.getCommitTimestamp() > txn.getBeginTimestamp()) { // bail out
+                throwWriteWriteConflict(txn, cell, dataTxnId);
+            }
+        }
+        if(!activeTx.isEmpty()) {
+            LOG.warn("transaction " + txn + " conflicts with " + activeTx.toString() + " on table " + table.getTableName() + " on row " + cell);
+        }
+        return Longs.toArray(activeTx);
+    }
+
+    private boolean shouldIgnoreConflicts(Partition table) throws IOException {
+        return "SYSSTATEMENTS".equals(table.getDescriptor().getValue("tableDisplayName"))
+                && (table.getDescriptor().getValue("schemaDisplayName") == null
+                || table.getDescriptor().getValue("schemaDisplayName").equals("SYS"));
+    }
```

Comment:
Did you change the SYSSTATEMENTS conflict avoidance in the end? The PR description still describes the old way.

Reply:
updated the PR description, thanks for checking it.